tests/e2e/network/ensureloadbalancer.go

```diff
@@ -139,14 +151,23 @@ var _ = Describe("Ensure LoadBalancer", func() {
 	})
 
 	It("should support BYO public IP", func() {
+		utils.Logf("Creating deployment %s", testDeploymentName)
+		deployment := createServerDeploymentManifest(testDeploymentName, labels)
```

Comment:
Why creating this?

Reply:
I moved createDeployment methods from BeforeEach into each test because for some test, I want to have a customizedd createDeployment. Here, for `should support BYO public IP` test, I just move the createDeployment and nothing more.

Reply:
Can we move this to WaitServiceExposureAndValidateConnectivity?

Reply:
I think we'd better keep it as it is because we have several different deployment and service manifests in e2e tests.

Reply:
If you want to customize the deployment, I think we can add variables in WaitServiceExposureAndValidateConnectivity. This would help reduce the duplicates in the code. WDYT?

Reply:
I will have a try.

Reply:
@nilo19 Updated. PTAL.